OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidICCAnalysis.scala

```diff
@@ -1120,28 +1118,22 @@ class AndroidICCAnalysis(val project: SomeProject) extends FPCFAnalysis {
     ): Integer = {
         val ssi = uri.indexOf(":")
         if (uri.substring(ssi).startsWith("//")) {
-            for (index ← ssi + 3 to uri.length) {
-                uri.charAt(index) match {
-                    case '/' | '\\' | '?' | '#' ⇒ return index;
-                    case _                      ⇒
-                }
-            }
-        }
-        uri.length
+            val sub = uri.substring(ssi + 3)
+            val indexList = List('/', '\\', '?', '#').map(_.toInt).map(sub.indexOf).filter(_ != -1)
+            if (indexList.nonEmpty) ssi + 3 + indexList.min
+            else uri.length
+        } else uri.length
     }
 
     def findPath(
         uri:   String,
         start: Integer
     ): String = {
-        var i = start + 1
-        while (i < uri.length) {
-            uri.charAt(i) match {
-                case '?' | '#' | '*' ⇒ return uri.substring(start + 1, i);
-                case _               ⇒ i += 1
-            }
-        }
-        uri.substring(start + 1)
+        val i = start + 1
+        val sub = uri.substring(i)
+        val indexList = List('?', '#', '*').map(_.toInt).map(sub.indexOf).filter(_ != -1)
```

Comment:
This still looks very suspicious. What is the map(_.toInt) doing there? Scala has an indexOf that takes a char (https://www.scala-lang.org/api/2.12.7/scala/collection/immutable/StringOps.html#indexOf(elem:Char):Int)
The combination of map, map, filter and min also looks inefficient. I suggest you use indexWhere to check for any of the three chars at once.
Also, instead of producing the substring sub, you can use the indexOf/indexWhere variants that use a `from` argument (take into account that this changes the semantics of the result though, it is then the index from the beginning of url, not the distance from `i`). You should then also get rid of `i` probably.